File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.cpp
class Solution {
public:
    int maxFreqSum(string s) {
        unordered_map<char,int>v;
        unordered_map<char,int>c;
        int n=s.length();
        int cf=0;
        int vf=0;
        for(auto &i:s)
        {
            if(i=='a'||i=='e'||i=='i'||i=='o'||i=='u')
            v[i]++;
            else
            c[i]++;
        }
        for(auto &i:v)
        {
            if(i.second>vf)
            vf=i.second;
        }
        for(auto &i:c)
        {
            if(i.second>cf)
            cf=i.second;
        }
        return (vf+cf);
        
    }
};

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.py
class Solution:
    def maxFreqSum(self, s: str) -> int:
        n = len(s)
        cv = 0   # max vowel freq
        cc = 0   # max consonant freq
        freq = {}

        vowels = {'a', 'e', 'i', 'o', 'u'}  

        for ch in s:
            if ch not in freq:
                freq[ch] = 0
            freq[ch] += 1
            if ch in vowels:
                cv = max(cv, freq[ch])
            else:
                cc = max(cc, freq[ch])

        return cc + cv


        